File: wrappers/test/js_common/test/did/did_key.test.ts
import { DIDKey, InMemKms, KeyType } from "equs-sdk";

describe("did:key: ", () => {
  it("generate DID with InMem KeyHandle", async () => {
    const kms = new InMemKms();
    const kid = await kms.create(KeyType.P256);
    const keyHandle = await kms.get(kid);

    const didKey = new DIDKey();

    const did = didKey.generate(keyHandle);

    expect(did.startsWith("did:key")).toBeTruthy();
  });

  it("generate DID with NOT an instance of InMem KeyHandle", async () => {
    const kms = new InMemKms();
    const kid = await kms.create(KeyType.P256);
    const keyHandle = await kms.get(kid);

    const didKey = new DIDKey();

    const did = didKey.generate({
      alg: keyHandle.alg,
      jwk: keyHandle.jwk,
      sign(payload: Uint8Array): Promise<Uint8Array> {
        return keyHandle.sign(payload);
      },
      verify(data: Uint8Array, signature: Uint8Array): Promise<void> {
        return keyHandle.verify(data, signature);
      },
    });

    expect(did.startsWith("did:key")).toBeTruthy();
  });
});

File: wrappers/uniffi/kotlin/src/test/kotlin/com/equs/DidKeyTest.kt
package com.equs

import com.equs.credentials.DidKey
import com.equs.credentials.InMemKms
import com.equs.credentials.KeyType
import com.equs.credentials.setJniLibPath
import kotlinx.coroutines.test.runTest
import org.junit.jupiter.api.BeforeAll
import kotlin.test.Test
import kotlin.test.assertContains

class DidKeyTest {
    companion object {
        @JvmStatic
        @BeforeAll
        fun setup() {
            setJniLibPath()
        }
    }

    @Test
    fun generateDid() = runTest {
        val kms = InMemKms()
        val kid = kms.create(KeyType.P256)
        val keyHandle = kms.get(kid)

        val didKey = DidKey()

        val did = didKey.generate(keyHandle)

        assertContains(did, "did:key")
    }
}


